sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -303,6 +308,20 @@ def delete_entity(
                 :dedent: 8
                 :caption: Deleting an entity to a Table
         """
+        try:
+            entity = kwargs.pop('entity', None)
+            if not entity:
+                entity = args[0]
+            partition_key = entity['PartitionKey']
+            row_key = entity['RowKey']
+        except (TypeError, IndexError):
+            partition_key = kwargs.pop('partition_key', None)
+            if not partition_key:
+                partition_key = args[0]
+            row_key = kwargs.pop('row_key', None) or args[1]
+            if not row_key:
```

Comment:
The `or` on line 321 should have made this `if` redundant?

Reply:
Good catch, I fixed line 321 to not have the `or`. It has to be an `if` because if the row_key is in the `**kwargs` it won't be in the `*args` which will throw an `IndexError`.